heap_sort/heap_sort.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * C program to Implement Heap sort.
+ * Time complexity: Average = O( n log( n ) ), worst case complexity = O( n log( n ) ).
+ * Space complexity: O(1).
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+// root = index root of the subtree, a is an array, heapsize is size of heap
+void  max_heapify(int a[], int root, int heapsize) {
+    int largest;
+    int l = (2 * root) + 1;  // left child
+    int r = (2 * root) + 2;  // Right child
+    // Check if left child is larger than root.
+    if ((l <= heapsize) && (a[l] > a[root]))
```

Comment:
There seems to be a big mistake here `l <= heapsize` isn't correct, because you are saying `heapsize` is the size of the heap but think while constructing max_heap initially, you would pass it the size of heap but actually largest accessible index is `heapsize-1`.
But I see in your main function you have passed `size-1` as size of heap which I think isn't the right way. Views? @singhpratyush